File: app/src/main/java/com/chatterbox/chatterbox/SplashScreen.java
package com.chatterbox.chatterbox;

import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ImageView;
import android.widget.TextView;

/**
 * Project: ChatterBox
 * Package: com.chatterbox.chatterbox
 * Created by lusinabrian on 13/08/16 at 07:47
 * <p/>
 * Description: Handles the SplashScreen Activity, which is the first thing to open when app starts
 */
public class SplashScreen extends AppCompatActivity{
    private final String SPLASHSCREEN_TAG = SplashScreen.class.getSimpleName();
    private TextView appName, appTag;
    private ImageView appIcon;
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        /*request full screen*/
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);

        setContentView(R.layout.splashscreen_layout);
        initUICtrls();
        /*Sets the timer*/
        Thread timer = new Thread(){
            @Override
            public void run(){
                try{
                    sleep(3000);
                    /*open sign in screen*/
                    Intent openMain = new Intent(SplashScreen.this, SignInActivity.class);
                    startActivity(openMain);
                }catch(InterruptedException ie){
                    ie.printStackTrace();
                    Log.d(ie.toString(),SPLASHSCREEN_TAG);
                }
            }
        };
        timer.start();
    }
    /**initialized the UI controls for this splash screen*/
    public void initUICtrls(){
        appIcon = (ImageView)findViewById(R.id.appicon_splash);
        appName = (TextView)findViewById(R.id.appname_splash);
        appTag = (TextView)findViewById(R.id.apptag_splash);
        //set the fonts
        String fontPath = "fonts/roboto_rediumitalic.ttf";
        Typeface typeface = Typeface.createFromAsset(getAssets(), fontPath);
        appName.setTypeface(typeface);
    }
    /*kill this splash screen to save memory*/
    @Override
    protected void onPause(){
        super.onPause();
        finish();
    }
/*CLASS END*/
}

File: app/src/main/java/com/chatterbox/chatterbox/ui/entry/splash/SplashScreen.kt
package com.chatterbox.chatterbox.ui.entry.splash

import android.content.Intent
import android.graphics.Typeface
import android.os.Bundle
import android.util.Log
import android.view.Window
import android.view.WindowManager
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.ImageView
import android.widget.TextView
import butterknife.ButterKnife

import com.chatterbox.chatterbox.R
import com.chatterbox.chatterbox.ui.auth.AuthActivity
import com.chatterbox.chatterbox.ui.auth.login.LogSignActivity
import com.chatterbox.chatterbox.ui.base.BaseActivity
import javax.inject.Inject

/**
 * Project: ChatterBox
 * Package: com.chatterbox.chatterbox
 * Created by lusinabrian on 13/08/16 at 07:47
 * Description: Handles the SplashScreen Activity, which is the first thing to open when app starts
 */
class SplashScreen : BaseActivity(), Animation.AnimationListener, SplashView {

    private val TAG = SplashScreen::class.java.simpleName
    
    private lateinit var animFadeIn: Animation
    private lateinit var appName: TextView
    private lateinit var appTag: TextView
    private lateinit var appIcon: ImageView
    
    @Inject
    lateinit var splashPresenter : SplashPresenter<SplashView>

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        /*request full screen*/
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)

        setContentView(R.layout.splashscreen_layout)
        
        activityComponent.inject(this)
        
        setUnbinder(ButterKnife.bind(this))
        
        splashPresenter.onAttach(this)
        
        setUp()
    }

    override fun setUp() {
        appIcon = findViewById(R.id.appicon_splash) as ImageView
        appName = findViewById(R.id.appname_splash) as TextView
        appTag = findViewById(R.id.apptag_splash) as TextView

        //set the fonts
        val fontPath = "fonts/roboto_rediumitalic.ttf"
        val typeface = Typeface.createFromAsset(assets, fontPath)

        appName.typeface = typeface

        /*add animation to application tag*/
        appTag.animation = animFadeIn

        //load animations
        animFadeIn = AnimationUtils.loadAnimation(applicationContext, R.anim.fade_in)
        //add animation listener
        animFadeIn.setAnimationListener(this)

        /*Sets the timer*/
        val timer = object : Thread() {
            override fun run() {
                try {
                    Thread.sleep(3000)
                    /*open sign in screen*/
                    val openMain = Intent(this@SplashScreen, AuthActivity::class.java)
                    startActivity(openMain)
                } catch (ie: InterruptedException) {
                    ie.printStackTrace()
                    Log.d(ie.toString(), TAG)
                }

            }
        }
        timer.start()
    }
    
    /*kill this splash screen to save memory*/
    override fun onPause() {
        super.onPause()
        finish()
    }

    override fun onAnimationStart(animation: Animation) {

    }

    override fun onAnimationEnd(animation: Animation) {

    }

    override fun onAnimationRepeat(animation: Animation) {

    }

}